benzina/src/json/mod.rs

```diff
@@ -0,0 +1,126 @@
+use std::fmt::Debug;
+
+use diesel::{
+    deserialize::{FromSql, FromSqlRow},
+    expression::AsExpression,
+    pg::{Pg, PgValue},
+    serialize::ToSql,
+    sql_types,
+};
+use serde::{Serialize, de::DeserializeOwned};
+
+use crate::json::convert::{sql_deserialize, sql_serialize};
+
+pub(crate) mod binary;
+pub(crate) mod convert;
+pub(crate) mod nullable;
+
+/// A diesel [`Json`] serialization and deserialization
+/// wrapper
+///
+/// For a JSON column in diesel, you would use the [`serde_json::Value`] model field.<br>
+/// If you need the control over the JSON structure you need to create your custom type that implements [`FromSql`] and [`ToSql`].
+///
+/// This wrapper allows to control the JSON structure without reimplementing that traits for each JSON column.
```

Comment:
This sounds clearer to me:


```suggestion
/// Diesel only implements [`FromSql`] and [`ToSql`] for [`serde_json::Value`], making
/// it hard to deal with `JSON` columns. This type implements [`FromSql`] and [`ToSql`]
/// for any type that implements [`Deserialize`] and [`Serialize`] respectively.
```